File: Assets/Bundles/Server/Resources/TS/Services/Global/Entity/EntityCommand.lua
-- Compiled with unity-ts v2.1.0-75
local Flamework = require("Shared/rbxts_include/node_modules/@easy-games/flamework-core/out/init").Flamework
local EntityPrefabType = require("Shared/TS/Entity/EntityPrefabType").EntityPrefabType
local ChatCommand = require("Server/TS/Services/Global/Chat/Commands/ChatCommand").ChatCommand
local EntityCommand
do
	local super = ChatCommand
	EntityCommand = setmetatable({}, {
		__tostring = function()
			return "EntityCommand"
		end,
		__index = super,
	})
	EntityCommand.__index = EntityCommand
	function EntityCommand.new(...)
		local self = setmetatable({}, EntityCommand)
		return self:constructor(...) or self
	end
	function EntityCommand:constructor()
		super.constructor(self, "entity", { "e" })
	end
	function EntityCommand:Execute(player, args)
		local entityService = Flamework.resolveDependency("Bundles/Server/Services/Global/Entity/EntityService@EntityService")
		if not player.Character then
			return nil
		end
		local pos = player.Character.gameObject.transform.position
		entityService:SpawnEntityForPlayer(nil, EntityPrefabType.HUMAN, pos)
	end
end
return {
	EntityCommand = EntityCommand,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Typescript~/src/Server/Services/Global/Entity/EntityCommand.ts
import { Dependency } from "@easy-games/flamework-core";
import { EntityPrefabType } from "Shared/Entity/EntityPrefabType";
import { Player } from "Shared/Player/Player";
import { ChatCommand } from "../../../Commands/ChatCommand";
import { EntityService } from "./EntityService";

export class EntityCommand extends ChatCommand {
	constructor() {
		super("entity", ["e"]);
	}

	public Execute(player: Player, args: string[]): void {
		const entityService = Dependency<EntityService>();

		if (!player.Character) return;

		const pos = player.Character.gameObject.transform.position;
		entityService.SpawnEntityForPlayer(undefined, EntityPrefabType.HUMAN, pos);
	}
}
